src/commands/CmdModify.cpp

```diff
@@ -116,6 +116,11 @@ void CmdModify::checkConsistency(Task &before, Task &after) {
 
   if (before.has("recur") && (!after.has("recur") || after.get("recur") == ""))
     throw std::string("You cannot remove the recurrence from a recurring task.");
+
+  if ((before.getStatus() == Task::pending) && (after.getStatus() == Task::pending) &&
+      (after.get("end") != ""))
+    throw format("Could not modify task {1}. You cannot set an end date on a pending task.",
+                 before.identifier(true));
```

Comment:
To answer your [question from the issue](https://github.com/GothenburgBitFactory/taskwarrior/issues/3584#issuecomment-2347969857), in this context I would prefer a warning since tasks might be `pending` but have an `end` from some other cause, and this will cause Taskwarrior to return an error even when the rnuning `task` command did not modify either of these properties.

That said, as you have seen there are lots of other `throw`s in this function, and I think adding another is the right course. We can think about converting them to warnings later.

Reply:
Oh, sorry, I admit I copy-pasted this comment from the other PR and didn't notice it's now in `CmdModify`! I think this is a much better fix :)